app/forms/find/subjects_form.rb

```diff
@@ -4,7 +4,15 @@ class SubjectsForm
 
     attr_accessor :subjects, :age_group
 
-    validates :subjects, presence: true
+    validate :subjects_have_been_selected
+
+    def subjects_have_been_selected
+      if subjects.blank? && age_group.present?
+        errors.add(:subjects, :"#{age_group}_subject")
+      elsif subjects.blank?
+        errors.add(:subjects, :blank)
+      end
```

Comment:
I don't think it is possible scenario. Can we remove?

Reply:
From a scenario perspective I agree, but from a philosophical perspective we get 
```
 [6] pry(main)> s = Find::SubjectsForm.new
=> #<Find::SubjectsForm:0x0000000110919e78>
[7] pry(main)> s.valid?
=> true
```

Reply:
Can we add a test for it then its not obvious, why its there and the lack of tests doesn't help .

Reply:
We have a test for it? 
```
it "is not valid when subject codes are not present" do
        form = described_class.new

        expect(form.valid?).to be(false)
      end
```

Reply:
As far as I can, see we only use it here:

https://github.com/DFE-Digital/publish-teacher-training/blob/81b1d1585b35b798cc7b418258f4eccfd4d3f0a9/app/controllers/find/search/subjects_controller.rb#L9

https://github.com/DFE-Digital/publish-teacher-training/blob/81b1d1585b35b798cc7b418258f4eccfd4d3f0a9/app/controllers/find/search/subjects_controller.rb#L13


Can we fix this class altogether, as the both our discussion and tests highlight it tech debt rolling forward with us, and distracting us, so can we boy scout :

1. remove
```
     elsif subjects.blank?
        errors.add(:subjects, :blank)
      end
```

2. amend
```
   #  attr_accessor :subjects, :age_group
   # to
    attr_reader :subjects, :age_group
```

3. add
```
    def initialize(subjects:, age_group:)
      @subjects = subjects
      @age_group = age_group
    end
```